pkg/controllers/work/patch_util.go

```diff
@@ -105,8 +106,22 @@ func setModifiedConfigurationAnnotation(obj runtime.Object) error {
 		return err
 	}
 	// set the last applied annotation back
-	annotations[lastAppliedConfigAnnotation] = string(modified)
-	return metadataAccessor.SetAnnotations(obj, annotations)
+	annotations[LastAppliedConfigAnnotation] = string(modified)
+	if err = metadataAccessor.SetAnnotations(obj, annotations); err != nil {
+		return err
+	}
+	annotations, err = metadataAccessor.Annotations(obj)
+	if err != nil {
+		return err
+	}
+	if err := validation.ValidateAnnotationsSize(annotations); err != nil {
+		klog.InfoS(fmt.Sprintf("setting last applied config annotation to empty, %s", err))
```

Comment:
```suggestion
		klog.klog.V(2).InfoS(fmt.Sprintf("setting last applied config annotation to empty, %s", err))
```
need to add the verbosity level

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md#what-method-to-use